MMapReadBufferFromFileDescriptor use getpagesize

kitaisreal opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

https://man7.org/linux/man-pages/man2/getpagesize.2.html

Changelog category (leave one):
- Not for changelog (changelog entry is not required)

kitaisreal committed:
MMapReadBufferFromFileDescriptor use getpagesize

alexey-milovidov reviewed (commented):
I just read that
```
NOTES
       Portable applications should employ sysconf(_SC_PAGESIZE) instead of getpagesize():

           #include <unistd.h>
           long sz = sysconf(_SC_PAGESIZE);

       (Most systems allow the synonym _SC_PAGE_SIZE for _SC_PAGESIZE.)

       Whether  getpagesize()  is  present as a Linux system call depends on the architecture.  If it is, it returns the kernel symbol PAGE_SIZE, whose value depends on the architecture and machine model.  Generally, one uses binaries that
       are dependent on the architecture but not on the machine model, in order to have a single binary distribution per architecture.  This means that a user program should not find PAGE_SIZE at compile time from a header file, but use an
       actual  system  call,  at  least for those architectures (like sun4) where this dependency exists.  Here glibc 2.0 fails because its getpagesize() returns a statically derived value, and does not use a system call.  Things are OK in
       glibc 2.1.
```

We should search for all instances of `getpagesize`, `sysconf(_SC_PAGESIZE)` and `4096` in code. And carefully unify them.

kitaisreal committed:
Unified usages of getPageSize

1. Introduced getPageSize function.
2. Replaced usages of getpagesize with getPageSize function.
3. Replaced usages of sysconf(_SG_PAGESIZE) with getPageSize function.

kitaisreal committed:
Replaced usages of 4096 constant with getPageSize

kitaisreal commented:
@alexey-milovidov thanks for review. I introduces getPageSize function and unified usages of getpagesize and sysconf(_SC_PAGESIZE). 
I also replaced most usages of constant 4096 with getPageSize, in places where it probably was used in assumption of memory page size. Probably in some places I should not replace it, please recheck.
There is also a lot of places where compile time constant 4096 is used as default argument in allocators for example like initial size, should it also be unified ?

kitaisreal committed:
Removed getPageSize from small files

kitaisreal committed:
Style check fix

kitaisreal committed:
Move getPageSize in common

kitaisreal committed:
Updated ya.make
